tools/tidy

```diff
@@ -76,49 +68,11 @@ if [ "$perltidy_version_found" != "$perltidy_version_expected" ]; then
     fi
 fi
 
-find-files() {
-    local files=()
-    [[ -d script ]] && files+=(script/*)
-    # shellcheck disable=SC2207
-    files=($(file --mime-type -- * "${files[@]}" | (grep text/x-perl || true) | awk -F':' '{ print $1 }'))
-    files+=('**.p[ml]' '**.t')
-    if $only_changed; then
-        git status --porcelain "${files[@]}" | awk '{ print $2 }'
-    else
-        git ls-files "${files[@]}"
-    fi
-}
-
 # go to caller directory
 cd "$(dirname "$0")/.."
 
 # just to make sure we are at the right location
 test -e tools/tidy || exit 1
 
-cleanup
-
-if [[ -n "$filename" ]]; then
-   if [[ -f "$filename" ]]; then
-       # shellcheck disable=SC2086
-       perltidy $PERLTIDY_ARGS "$filename"
-   else
-       echo "\"$filename\" is not a valid file, please provide only one file"
-       exit 22
-   fi
-else
-    # shellcheck disable=SC2086
-    find-files | xargs perltidy $PERLTIDY_ARGS
-fi
-
-(git ls-files --others "*.tdy" || find . -name "*.tdy") | while read -r file
-do
-    if diff -u "${file%.tdy}" "$file"; then
-        continue
-    fi
-    if [[ -n "$check" ]]; then
-        echo "RUN tools/tidy script before checkin"
-        exit 1
-    else
-        mv -v "$file" "${file%.tdy}"
-    fi
-done
+# shellcheck disable=SC2086
+tidyall $selection $args $filename
```

Comment:
When trying to invoke it with a parameter I get:

```
tools/tidy --check
must pass -a/--all, -s/--svn, -g/--git, -p/--pipe, or filename(s)
Usage: tidyall [options] [file] ...
```

I see you've just updated the PR. I would have suggested the following otherwise:

```suggestion
tidyall $args ${filename:-"-a"}
```

Reply:
Good idea. I have additionally changed it so that all three "filename", "--all", "--git" work now. If one or multiple filenames are given then this is taken, otherwise if "--only-changed" selected then "--git" is supplied, otherwise "--all"